modules/kafka/src/main/java/org.testcontainers.containers/KafkaContainer.java

```diff
@@ -0,0 +1,61 @@
+package org.testcontainers.containers;
+
+import org.testcontainers.utility.Base58;
+
+import java.util.stream.Stream;
+
+public class KafkaContainer extends GenericContainer<KafkaContainer> {
+
+    public static final int KAFKA_PORT = 9092;
+
+    public static final int ZOOKEEPER_PORT = 2181;
+
+    protected SocatContainer proxy;
+
+    public KafkaContainer() {
+        this("4.0.0");
+    }
+
+    public KafkaContainer(String confluencePlatformVersion) {
+        super("confluentinc/cp-kafka:" + confluencePlatformVersion);
+
+        withNetwork(Network.newNetwork());
+        withNetworkAliases("kafka-" + Base58.randomString(6));
+        withExposedPorts(KAFKA_PORT, ZOOKEEPER_PORT);
+
+        withEnv("KAFKA_BROKER_ID", "1");
+        withEnv("KAFKA_LISTENERS", "PLAINTEXT://0.0.0.0:9092,BROKER://127.0.0.1:9093");
+        withEnv("KAFKA_LISTENER_SECURITY_PROTOCOL_MAP", "BROKER:PLAINTEXT,PLAINTEXT:PLAINTEXT");
+        withEnv("KAFKA_INTER_BROKER_LISTENER_NAME", "BROKER");
```

Comment:
This line makes the whole thing work :D Otherwise, Kafka would try to communicate with itself with host's port  (yes, even in 1-node scenario (sic!) )

Reply:
I'm not directly familiar with Kafka or the 7 environment variables that are being set in this method. Is it possible that someone might want to use alternative values for some of these? If so, they could happily subclass and re-set the env vars they want, so that's OK.

However, would it be worth having a comment in the code (similar to your Github comment!) to say which settings are just critical to getting Kafka to work in the Dockerised environment and shouldn't be changed?